geonode/documents/renderers.py

```diff
@@ -58,28 +58,23 @@ def render_document(document_path, extension="png"):
 
     # workaround: https://github.com/dagwieers/unoconv/issues/167
     # first convert a document to PDF and continue
-    if extension == "pdf":
-        temp_path = document_path
-    elif guess_mimetype(document_path) == 'application/pdf':
-        temp_path = document_path
-    else:
-        temp = render_document(document_path, extension="pdf")
-        temp_path = temp.name
+    dispose_input = False
+    if extension != "pdf" and guess_mimetype(document_path) != 'application/pdf':
+        document_path = render_document(document_path, extension="pdf")
+        dispose_input = True
 
     # spawn subprocess and render the document
-    output = NamedTemporaryFile(suffix='.{}'.format(extension))
+    output_path = None
     if settings.UNOCONV_ENABLE:
         timeout = None
+        _, output_path = tempfile.mkstemp(suffix=".{}".format(extension))
```

Comment:
Why use `mkstemp` instead of `NamedTemporaryFile` as in the original code? I don't see the improvement?

Reply:
`mkstemp` provides more control of when the file gets cleaned up. This approach was the only way I could get the large temporary image generated by `unoconv` to persist long enough to generate thumbnail content. These calls are equivalent so long as the file gets deleted at the end.